client/src/route/dashboard/Dashboard.tsx

```diff
@@ -1,13 +1,97 @@
+/* eslint-disable no-console */
 /*
 source: https://github.com/mui/material-ui/tree/v5.10.0/docs/data/material/getting-started/templates/dashboard
 */
 import * as React from 'react';
 import Grid from '@mui/material/Grid';
 import Paper from '@mui/material/Paper';
+
+import { useEffect } from 'react';
+import { getCodeParam, getGitLabAccessToken, checkTokenValidity, resetToken} from '../../util/authentication';
+import fetchGitLabGraphQLQuery from '../../util/graphQL';
+// import fetchProtectedApi from '../../util/api';
+
 import Chart from '../history/Chart';
 import RecentRuns from '../history/RecentRuns';
 
+
+const CLIENT_ID = window.env.REACT_APP_CLIENT_ID;
+const CLIENT_SECRET = window.env.REACT_APP_CLIENT_SECRET;
+const REDIRECT_URL = window.env.REACT_APP_REDIRECT_URL;
+const GITLAB_OAUTH_TOKEN = window.env.REACT_APP_GITLAB_OAUTH_TOKEN;
+const GITLAB_API_GRAPHQL = window.env.REACT_APP_GITLAB_API_GRAPHQL;
+
 function DashboardContent() {
+  useEffect(() => {
+    const codeParam = getCodeParam();
+    if (localStorage.getItem('expiresIn')) {
+      if (checkTokenValidity() === true) {
+        fetchGitLabGraphQLQuery(localStorage.getItem('accessToken'), `
+        {
+          currentUser {
+            id
+            name
+            username
+            email
+          }
+        }
+      `, GITLAB_API_GRAPHQL).then((data) => {
+          console.log('Current user: ', data?.data?.currentUser.name, data?.data?.currentUser.username);
```

Comment:
please comment out console.log() statements. We can add [winston logger](https://github.com/winstonjs/winston) later.